### PR TITLE
Worker Thread: Member Oper (V1/V2)

### PR DESCRIPTION
**Pull request by hthompson6**
Updates the operating statuses of service group members for v1 and v2

### CODE REVIEW

**Comment by hthompson6** [comment]
check

File: a10_neutron_lbaas/tests/unit/v1/test_handler_member.py [hunk 1]
```diff
@@ -113,3 +116,12 @@ def test_delete_count_gt_one(self):
         pool_name = self.a.member._pool_name(None, m['pool_id'])
         self.a.last_client.slb.service_group.member.delete.assert_called_with(
             pool_name, name, m['protocol_port'])
+
+    def test_updating_oper_stats(self):
+        status_check.status_update_v1(self.a)
+        self.a.last_client.slb.service_group.member.get_oper.mock_calls[0].assert_called_with(
```
Comment (mdurrant-b3) [inline a10_neutron_lbaas/tests/unit/v1/test_handler_member.py]: If `get_oper` has no `mock_calls` made to it, this line will fall on it's face.  This needs a preliminary length check and/or ternary operator assignment.

File: a10_neutron_lbaas/tests/unit/v2/test_handler_member.py [hunk 2]
```diff
@@ -119,3 +120,12 @@ def test_delete_count_gt_one(self):
 
         self.a.last_client.slb.service_group.member.delete.assert_called_with(
             m.pool_id, name, m.protocol_port)
+
+    def test_updating_oper_stats(self):
+        status_check.status_update_v2(self.a)
+        self.a.last_client.slb.service_group.member.get_oper.mock_calls[0].assert_called_with(
```
Comment (mdurrant-b3) [inline a10_neutron_lbaas/tests/unit/v2/test_handler_member.py]: This is redundant given the test above.  I try to stick to 1 assertion per test - exceptions to this are: assert return value of call is not None, then assert what I'm actually testing.  This makes the test fail due to the value not being returned and provides clearer failure output.

File: a10_neutron_lbaas/tests/unit/test_base.py [hunk 3]
```diff
@@ -52,7 +58,14 @@ def reset_mocks(self):
...
 class FakeA10OpenstackLBV1(FakeA10OpenstackLB, a10_os.A10OpenstackLBV1):
-    pass
+    def __init__(self, openstack_driver, **kw):
+        super(FakeA10OpenstackLBV1, self).__init__(
+            openstack_driver,
+            neutron_hooks_module=mock.MagicMock(),
+            **kw)
+        pool = v1_objs.FakePool()
+        pool.members = [v1_objs.FakeMember()]
+        self.openstack_driver.plugin.get_pools = lambda mem: [pool]
```
Comment (Cedev) [inline a10_neutron_lbaas/tests/unit/test_base.py]: This looks like setup that belongs in a test.

**Comment by hthompson6** [comment]
check

**Comment by dougwig** [comment]
Continue as a working environment for testing background functions, but we'll punt this direct to an async daemon instead of the pain of threads.

**Pull request closed without merging**